packages/availity-reactstrap-validation/src/AvStore.js

```diff
@@ -0,0 +1,364 @@
+  */
+  inputValidations = {}; // store the inputs provided validations object
+
+  allInputValidations = {}; // store the merged defaults and defined validations
+  watchers = {}; // cache which validators are watching each input
+
+  /*
+    TODO: determine what is needed all the time or not
+    stateObj = {
+      name: string (defaults to id),
+      value: store the value of the input
+      dirty: T/F, (if value has been changed by the input)
+      bad: T/F (if input even has error)
+      pending: T/F, (if waiting for async validations)
+      valid: T/F, if has any errors
+      error: {
+        errorKey: T/errorMessage
+      }
+    }
+  */
+  inputs = {};
+  /*
+    inputId: stateObj
+  */
+  subscribers = {}; // store all subscribers
+
+  constructor(options = {}) {
+    const {
+      values,
+      validators,
+      errorMessages,
+      validations,
+    } = options;
+    this.values = values;
+    this.validators = validators;
+    this.errorMessages = errorMessages;
+    if (validations) {
+      this.updateInputValidations(validations);
+    }
+  }
+
+  subscribe(name, subscriber, options) {
+    if (!name) {
+      throw new Error('name is required to register');
+    }
+
+    if (!isString(name)) {
+      throw new Error('name must be a string');
+    }
+
+    if (name[0] === '.') {
+      throw new Error('name cannot start with .');
+    }
+
+    if (this.subscribers[name]) {
+      throw new Error(`${name} is already subscribed`);
+    }
+
+    if (!isFunction(subscriber)) {
+      throw new Error('must pass in subscribing function');
+    }
+
+    this.subscribers[name] = subscriber;
+    this.inputs[name] = {
+      touched: false,
+      dirty: false,
+      bad: false,
+      valid: true,
+      pending: false,
+    };
+
+    if (options) {
+      const {state, validations, value} = options;
+      if (validations) {
+        this.updateInputValidations(validations, name);
+      }
+      if (state) {
+        Object.assign(this.inputs[name], state);
+      }
+      if (value) {
+        Object.assign(this.inputs[name], { value });
+      }
+    }
+
+    return () => {
+      delete this.subscribers[name];
+      // update validations after subscribers gone to trigger removal
+      this.updateInputValidations({}, name);
+      // remove inputs last because watchers is needed for validations
+      delete this.inputs[name];
+    };
+  }
+
+  notifySubscriber = (inputId) => {
+    // TODO: determine what information to provide on notification
+    this.subscribers[inputId]();
+  }
+
+  updateValues = (values) => {
+    this.values = values;
+    Object.keys(this.inputs).forEach(inputId => {
+      const newValue = this.getDefaultValue(inputId);
+      const currentValue = this.getValue(inputId);
+      if (newValue !== currentValue) {
+        this.updateInput(inputId, newValue, true);
+      }
+    });
+  }
+
+  updateValue = (inputId, value, validate) => {
+    this.updateInput(inputId, {value}, validate);
```

Comment:
this will call `this.updateInput` with the second argument being an object with `value`, in `updateValues` (above) it calls with potentially a string, (what is stored in `this.values` / what does `this.getDefaultValue` return?)

Reply:
Values is where the store level defaults are stored (the forms model). 
perhaps it would be easier if we used model instead of values?

so updateValue and updateInput will update the input's state object. updateValues will update the model and propagate those changes to the inputs value. 

Lastly I did just notice that updateValues should call updateValue not updateInput inside